File: python/medium/bubble_sort.py
from typing import List

def bubble_sort(l: List[int]) -> List[int]:
    a = l[:]
    is_finish = False

    while not is_finish:
        for i in range(1, len(a)):
            prev = a[i-1]
            current = a[i]
            if prev > current:
                a[i-1] = current
                a[i] = prev
            elif current == len(a) - 1 and prev < current:
                is_finish = True
    return a
        
print(bubble_sort([3,1,4,2]))

File: ts/medium/bubbleSort.ts
function bubbleSort(arr: number[]): number[] {
  const copy = [...arr];
  let isFinish = false;

  while (!isFinish) {
    for (let i = 1; i < copy.length; i++) {
      const prev = copy[i - 1];
      const current = copy[i];
      if (prev > current) {
        copy[i - 1] = current;
        copy[i] = prev;
      } else if (current === copy.length - 1 && prev < current) {
        isFinish = true;
      }
    }
  }
  return copy;
}
console.log(bubbleSort([3,1,4,2]))